src/csv_upload/csvUpload.test.js

```diff
@@ -0,0 +1,143 @@
+      csv: 'asset_manager_id,party_id,transaction_id'
+      +'\n'+'1,224,'
+      +'\n'+'1,2,3'
+      +'\n'+',4,5'
+  }
+
+  const data=[{
+      assetManagerId: 1,
+      partyId: 224,
+      transactionId: undefined
+  },
+  {
+      assetManagerId: 1,
+      partyId: 2,
+      transactionId: 3
+  },
+  {
+      assetManagerId: undefined,
+      partyId: 4,
+      transactionId: 5
+  }
+  ]
+
+  const param2={
+      csv: 'links.MyLink.0.linked_id,comments.MyComments.comment_value'
+      +'\n'+'1,PAID'
+      +'\n'+'2,REFUND'
+      +'\n'+'3,'
+      +'\n'+',PAID'
+  }
+
+  const data2=[
+      {
+        links:
+        {
+            MyLink:
+            [
+                {
+                  active: true,
+                  createdBy: "TEMP",
+                  createdTime: undefined,
+                  linkedId:  1,
+                  updatedBy: "TEMP",
+                  updatedTime: undefined,
+                  version: 1
+                }
+            ]
+        },
+        comments:
+        {
+            MyComments:
+            {
+              commentValue: "PAID",
+              createdBy: "TEMP",
+              updatedBy: "TEMP",
+              createdTime: undefined,
+              updatedTime: undefined,
+              version: 1,
+              active: true   
+            }
+        }
+
+      },
+      {
+        links:
+        {
+            MyLink:
+            [
+                {
+                    active: true,
+                    createdBy: "TEMP",
+                    createdTime: undefined,
+                    linkedId:  2,
+                    updatedBy: "TEMP",
+                    updatedTime: undefined,
+                    version: 1
+                }
+            ]
+        },
+        comments:
+        {
+            MyComments:
+            {
+              commentValue: "REFUND",
+              createdBy: "TEMP",
+              updatedBy: "TEMP",
+              createdTime: undefined,
+              updatedTime: undefined,
+              version: 1,
+              active: true  
+            }
+        }
+
+      },
+      {
+        links:
+        {
+            MyLink:
+            [
+                {
+                    active: true,
+                    createdBy: "TEMP",
+                    createdTime: undefined,
+                    linkedId:  3,
+                    updatedBy: "TEMP",
+                    updatedTime: undefined,
+                    version: 1
+                }
+            ]
+        },
+        comments: undefined
+      },
+      {
+        links: undefined,
+       
+        comments:
+        {
+            MyComments:
+            {
+              commentValue: "PAID",
+              createdBy: "TEMP",
+              updatedBy: "TEMP",
+              createdTime: undefined,
+              updatedTime: undefined,
+              version: 1,
+              active: true
+            }
+        }
+
+      }
+  ]
+
+  expect(csv.parseString(param2)).toEqual(data2);
```

Comment:
You also need to test that parsing `param` equals `data`. You should split this into 2 tests:

```javascript
describe('parseString', () => {
  it('should convert headings to  camelCase', () => {
    // Test the basic headings asset_manager_id, transaction_id etc.
    const param = //...
    const data = //...
    expect(csv.parseString(param)).toEqual(data)
  })

  it('should convert child headings to nested objects', () => {
    // Test the child headings links.MyLink.0.linked_id, comments.MyComment.comment_value etc.
    const param2 = //...
    const data2 = //...
    expect(csv.parseString(param2)).toEqual(data2)
  })
})
```